drgn/helpers/common/memory.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Memory
+------
+
+The ``drgn.helpers.memory`` module provides helpers for working with memory and addresses.
+"""
+
+import operator
+from typing import Optional, Union, overload
+
+import drgn
+from drgn import IntegerLike, Object, Program, SymbolKind
+from drgn.helpers.common.format import escape_ascii_string
+from drgn.helpers.linux.slab import find_containing_slab_cache
+
+__all__ = ("identify_address",)
+
+
+_SYMBOL_KIND_STR = {
+    SymbolKind.OBJECT: "object symbol",
+    SymbolKind.FUNC: "function symbol",
+}
+
+
+@overload
+def identify_address(addr: Object) -> Optional[str]:
+    """"""
+    ...
+
+
+@overload
+def identify_address(prog: Program, addr: IntegerLike) -> Optional[str]:
+    """
+    Identify an address (given as a :class:`.Program` and an integer) as a symbol or slab object.
+
+    This will currently identify the following types of addresses as follows:
+
+    * Symbols (e.g., addresses in functions or global variables): `symbol_kind: symbol_name+offset`.
+    * Slab objects: `slab object: slab_cache_name`.
+
+    This may learn to recognize other types of addresses in the future.
+
+    The address can be given as an :class:`~drgn.Object` or as a
+    :class:`~drgn.Program` and an integer.
+
+    :param addr: ``void *``
+    :return: Identity as string or ``None` if the address is unrecognized.
+    """
+    ...
+
+
+def identify_address(  # type: ignore  # Need positional-only arguments.
+    prog_or_addr: Union[Program, Object], addr: Optional[IntegerLike] = None
+) -> Optional[str]:
+    if addr is None:
+        assert isinstance(prog_or_addr, Object)
+        prog = prog_or_addr.prog_
+        addr = prog_or_addr
+    else:
+        assert isinstance(prog_or_addr, Program)
+        prog = prog_or_addr
+    addr = operator.index(addr)
+
+    if prog.flags & drgn.ProgramFlags.IS_LINUX_KERNEL:
+        # Linux kernel-specific identification:
+        slab_cache = find_containing_slab_cache(prog, addr)
+
+        if slab_cache.value_():
```

Comment:
You can check the truthiness of a `drgn.Object` directly:

```suggestion
        if slab_cache:
```

Reply:
Fixed